packages/shared/client/ReplayClient.ts

```diff
@@ -138,11 +149,19 @@ export class ReplayClient implements ReplayClientInterface {
     const { sessionId } = await client.Recording.createSession({ recordingId });
 
     this._sessionId = sessionId;
+    this.sessionWaiter.resolve(sessionId);
     this._threadFront.setSessionId(sessionId);
 
     return sessionId;
   }
 
+  async findKeyboardEvents(onKeyboardEvents: (events: keyboardEvents) => void) {
+    const sessionId = this.getSessionIdThrows();
```

Comment:
Since we're in an async function I think we should await the `sessionWaiter`, that would make this method (slightly) more robust. Same for all other uses of `getSessionIdThrows`.

Reply:
The only place this is called is inside of `setupApp()` in `ui/actions/app`, and that already does an explicit `ThreadFront.waitForSession().then()`.

Reply:
I'd still prefer the more robust version, so that we don't have to worry about it when calls to this method or any of the other methods using `getSessionIdThrows()` are moved around.

Reply:
Maybe we should just remove the session waiter.